chain/client/src/stateless_validation/state_witness_producer.rs

```diff
@@ -102,7 +103,12 @@ impl Client {
             // With stateless validation chunk producer uses recording reads when validating transactions.
             // The storage proof must be available here.
             transactions_storage_proof.ok_or_else(|| {
-                Error::Other("Missing storage proof for transactions validation".to_owned())
+                let msg = "Missing storage proof for transactions validation";
+                contract_violation!(
+                    target: "stateless_validation",
+                    "{}", msg
```

Comment:
You don't need to format this, `message` is conventionally the key for messages, so you could skip the formatting with `{}` if you named the variable `message`.

Reply:
Hm, right. The convention with `message` is for tracing, not for `debug_assert!`s. Panics and other std macros require a formatting pattern since edition 2021. This would work if it was coupled with [my other suggestion](https://github.com/near/nearcore/pull/10659#discussion_r1503413010), though.

Reply:
Got it, thanks!

Regarding your other suggestion: @shreyan-gupta [decided to give it a try](https://github.com/near/nearcore/pull/10673#discussion_r1504558034).